### PR TITLE
Add testing for client events

### PR DESCRIPTION
**Pull request by chatton**
## Description

Merge after #5682

Ref #2823

I had to do some refactoring of the `AssertEvents` function. The changes ensure that it is possible to pass either `res.Events` or `ctx.EventManager().Events().ToABCIEvents()` as expected events.

I opted to add some event testing into existing tests as they have quite an elaborate set up, and it decided it would be cleaner to just add additional assertions into those tests.

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to be used for the changelog entry in the PR description for review.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.
- [ ] Review `Codecov Report` in the comment section below once CI passes.

### CODE REVIEW

**Review by damiannolan — APPROVED** [review]
Awesome! 😎 

Thanks for picking these events tests up! Much appreciated

File: testing/events.go [hunk 1]
```diff
@@ -231,3 +225,39 @@ func AssertEvents(
...
+		return len(expectedEvent.Attributes) == len(actualEvent.Attributes)-1
+	}
+
+	return len(expectedEvent.Attributes) == len(actualEvent.Attributes)
+}
+
+// containsAttribute returns true if the given key/value pair is contained in the given attributes.
+// NOTE: this ignores the indexed field, which can be set or unset depending on how the events are retrieved.
+func containsAttribute(attrs []abci.EventAttribute, key, value string) bool {
+	for _, attr := range attrs {
```
Comment (colin-axner) [inline testing/events.go:247]: a slices.ContainsFn() can be used on these?

File: modules/core/02-client/keeper/events_test.go [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+	commitmenttypes "github.com/cosmos/ibc-go/v8/modules/core/23-commitment/types"
+	ibctm "github.com/cosmos/ibc-go/v8/modules/light-clients/07-tendermint"
+	ibctesting "github.com/cosmos/ibc-go/v8/testing"
+)
+
+func (suite *KeeperTestSuite) TestMsgCreateClientEvents() {
+	suite.SetupTest()
+	path := ibctesting.NewPath(suite.chainA, suite.chainB)
+
+	path.EndpointA.Counterparty.Chain.NextBlock()
```
Comment (colin-axner) [inline modules/core/02-client/keeper/events_test.go:16]: this line might not be necessary anymore

**Review by colin-axner — APPROVED** [review]
Nice, thanks for picking these up!

File: modules/core/02-client/keeper/events_test.go (hunk 2 above)
Comment (chatton) [reply modules/core/02-client/keeper/events_test.go:16]: looks like the test fails without this.

File: testing/events.go (hunk 1 above)
Comment (chatton) [reply testing/events.go:247]: nice suggestion we can use

```golang
	return slices.ContainsFunc(attrs, func(attr abci.EventAttribute) bool {
		return attr.Key == key && attr.Value == value
	})
```

**Pull request merged**